metricbeat/module/jolokia/_meta/Dockerfile

```diff
@@ -1,11 +1,39 @@
 # Tomcat is started to fetch Jolokia metrics from it
-FROM jolokia/java-jolokia:7
-ENV TOMCAT_VERSION 7.0.55
+FROM java:8-jdk-alpine
+
+ENV TOMCAT_VERSION 7.0.86
 ENV TC apache-tomcat-${TOMCAT_VERSION}
+ENV JOLOKIA_VERSION 1.5.0
+ENV PROXT_TARGET_TOMCAT_VERSION 9.0.7
+ENV PX_TC apache-tomcat-${PROXT_TARGET_TOMCAT_VERSION}
+
+RUN apk update;\
+    apk add curl
 
-HEALTHCHECK --interval=1s --retries=90 CMD curl -f curl localhost:8778/jolokia/
+HEALTHCHECK --interval=1s --retries=90 CMD curl -f localhost:8778/jolokia/
 EXPOSE 8778
-RUN wget -q http://archive.apache.org/dist/tomcat/tomcat-7/v${TOMCAT_VERSION}/bin/${TC}.tar.gz
-RUN tar xzf ${TC}.tar.gz -C /opt
 
-CMD env CATALINA_OPTS=$(jolokia_opts) /opt/${TC}/bin/catalina.sh run
+# Prepare a server where jolokia runs in proxy mode
+RUN wget http://archive.apache.org/dist/tomcat/tomcat-7/v${TOMCAT_VERSION}/bin/${TC}.tar.gz;\
+    tar xzf ${TC}.tar.gz -C /usr;\
+    sed -i -e 's/Connector port="8080"/Connector port="8778"/g' /usr/${TC}/conf/server.xml;\
+    wget http://central.maven.org/maven2/org/jolokia/jolokia-war/${JOLOKIA_VERSION}/jolokia-war-${JOLOKIA_VERSION}.war -O /usr/${TC}/webapps/jolokia.war
+
+# Set up a server to be connected in proxy mode
+RUN wget http://archive.apache.org/dist/tomcat/tomcat-9/v${PROXT_TARGET_TOMCAT_VERSION}/bin/${PX_TC}.tar.gz;\
+    tar xzf ${PX_TC}.tar.gz -C /usr;\
+    sed -i -e 's/Connector port="8/Connector port="18/g' /usr/${PX_TC}/conf/server.xml;\
+    sed -i -e 's/Server port="8005"/Server port="18005"/g' /usr/${PX_TC}/conf/server.xml
+
+# JMX setting to request authentication with remote connection
+RUN echo "monitorRole QED" >> /usr/lib/jvm/java-1.8-openjdk/jre/lib/management/jmxremote.password;\
+    echo "controlRole R&D" >> /usr/lib/jvm/java-1.8-openjdk/jre/lib/management/jmxremote.password;\
+    chmod 600 /usr/lib/jvm/java-1.8-openjdk/jre/lib/management/jmxremote.password
+
+# Start tomcat to accept JMX connection and enable jolokia proxy mode
+CMD env CATALINA_OPTS="$(jolokia_opts)\
+    -Dcom.sun.management.jmxremote.port=7091\
+    -Dcom.sun.management.jmxremote.ssl=false\
+    -Dcom.sun.management.jmxremote.authenticate=true"\
+    /usr/${PX_TC}/bin/catalina.sh run & \
+    env CATALINA_OPTS="-Dorg.jolokia.jsr160ProxyEnabled" /usr/${TC}/bin/catalina.sh run
```

Comment:
Would it be possible to run everything in an only tomcat?

Reply:
In order to check the connection in Proxy mode, Tomcat to be connected is added.
If you use in an only Tomcat, you will not know if you are connected to another Tomcat in the process of connecting in Proxy mode of the integration test.

Reply:
Ok, makes sense, maybe we could think in a two container scenario, but I think we can continue as is.

Reply:
Changed the connecting to another Tomcat to a pattern where jmx password authentication cannot be performed correctly.
Fix to be possible to run everything in an only Tomcat.